cmd/vic-machine/validate/validator.go

```diff
@@ -45,314 +51,577 @@ type Validator struct {
+func NewValidator(ctx context.Context, input *data.Data) (*Validator, error) {
+	var err error
 
-	v.TargetPath = input.URL.String()
-	vchConfig.Target = v.TargetPath
-	vchConfig.Insecure = input.Insecure
+	v := &Validator{}
+	v.Context = ctx
+	tURL := input.URL
 
-	if v.DatacenterName != "" {
-		v.ImageStorePath = fmt.Sprintf("/%s/datastore/%s", v.DatacenterName, input.ImageDatastoreName)
-		v.ExternalNetworkPath = fmt.Sprintf("/%s/network/%s", v.DatacenterName, input.ExternalNetworkName)
-		v.BridgeNetworkPath = fmt.Sprintf("/%s/network/%s", v.DatacenterName, input.BridgeNetworkName)
-		v.ManagementNetworkPath = fmt.Sprintf("/%s/network/%s", v.DatacenterName, input.ManagementNetworkName)
-		vchConfig.DatacenterName = v.DatacenterName
+	// default to https scheme
+	if tURL.Scheme == "" {
+		tURL.Scheme = "https"
 	}
 
-	v.BridgeNetworkName = input.BridgeNetworkName
-	v.ManagementNetworkName = input.ManagementNetworkName
-	vchConfig.ImageStoreName = input.ImageDatastoreName
-
-	if v.ClusterPath != "" {
-		vchConfig.ClusterPath = v.ClusterPath
+	// if they specified only an IP address the parser for some reason considers that a path
+	if tURL.Host == "" {
+		tURL.Host = tURL.Path
+		tURL.Path = ""
 	}
 
-	if err := v.validateConfiguration(input, vchConfig); err != nil {
-		return nil, err
+	sessionconfig := &session.Config{
+		Insecure: input.Insecure,
 	}
-	return vchConfig, nil
-}
 
-func (v *Validator) ParseComputeResourcePath(rp string) error {
-	resources := strings.Split(rp, "/")
-	if len(resources) < 2 || resources[1] == "" {
-		err := errors.Errorf("Could not determine datacenter from specified --compute-resource path: %s", rp)
-		return err
+	// only allow the datacenter to be specified in the taget url, if any
+	pElems := filepath.SplitList(tURL.Path)
+	if len(pElems) > 1 {
+		detail := "Target should specify only the datacenter in the path component (e.g. https://addr/datacenter) - resource pools or folders are separate arguments"
+		log.Error(detail)
+		return nil, errors.New(detail)
 	}
-	v.DatacenterName = resources[1]
-	v.ClusterPath = strings.Split(rp, "/Resources")[0]
 
-	if v.ClusterPath == "" {
-		err := errors.Errorf("Could not determine cluster from specified --compute-resource path: %s", rp)
-		return err
+	// if a datacenter was specified, set it
+	v.DatacenterName = tURL.Path
+	if v.DatacenterName != "" {
+		sessionconfig.DatacenterPath = v.DatacenterName
+		// path needs to be stripped before we can use it as a service url
+		tURL.Path = ""
 	}
-	v.ResourcePoolPath = rp
-	return nil
-}
 
-func (v *Validator) GetResourcePool(input *data.Data) (*object.ResourcePool, error) {
-	if err := v.ParseComputeResourcePath(input.ComputeResourcePath); err != nil {
+	sessionconfig.Service = tURL.String()
+
+	v.Session = session.NewSession(sessionconfig)
+	v.Session, err = v.Session.Connect(v.Context)
+	if err != nil {
 		return nil, err
 	}
-	if _, err := v.getConnection(input); err != nil {
+
+	v.Session, err = v.Session.Populate(ctx)
```

Comment:
This method will load default resources for cluster, datastore, etc. So you have to set value into session config, to avoid ambiguous.
I added one method getResources, to avoid session class, and that method  loads set resources only, so is OK to be invoked by all vic-machine commands. Is it good to invoke getResources here?